### PR TITLE
Fix minor model issues

### PR DESCRIPTION
**Pull request by ryanleary**
This PR fixes three issues:
1. squelches _some_ of the `flatten_parameters` warnings when on multi-gpu machines. There may still be an additional fix required (perhaps a call after the `DataParallel` call?).
2. Adds padding to the first CNN. It appears that utterances that start at the very beginning of an audio file are currently not well-handled due to the edge-effects. This has the added effect of further simplifying the time-alignment math.
3. Adds a "blacklist" of parameters to not load. This makes it possible to easily load older DeepSpeech models which had an architecture that included an unused BatchNorm on the first RNN.

### CODE REVIEW

File: model.py [hunk 1]
```diff
@@ -139,7 +142,16 @@ def load_model(cls, path, cuda=False):
         model = cls(rnn_hidden_size=package['hidden_size'], nb_layers=package['hidden_layers'],
                     labels=package['labels'], audio_conf=package['audio_conf'],
                     rnn_type=supported_rnns[package['rnn_type']])
+        # the blacklist parameters are params that were previous erroneously saved by the model
+        # care should be taken in future versions that if batch_norm on the first rnn is required
+        # that it be named something else
+        blacklist = ['rnns.0.batch_norm.module.weight', 'rnns.0.batch_norm.module.bias', 'rnns.0.batch_norm.module.running_mean', 'rnns.0.batch_norm.module.running_var']
```
Comment (ryanleary) [inline model.py]: I can be swayed to not include this and to instead write a script (to possibly check-in to a `utils` directory or something) to convert these old model formats.

Comment (SeanNaren) [reply model.py]: Won't previous models not work without these weights though? or atleast not give the same output?

Comment (ryanleary) [reply model.py]: We had a bug in the previous implementation, I think, where the batch rnn was added to the model, but never actually trained or used during decode. The parameters were just registered.

See below snippets:
1. https://github.com/SeanNaren/deepspeech.pytorch/blob/522ce348259507fafe07f0eb9028d91b1fa440d1/model.py#L45 In this implementation of `BatchRNN`, even if `batch_norm=False`, the BatchNorm module is added to this module. This registers the parameters.

2. https://github.com/SeanNaren/deepspeech.pytorch/blob/522ce348259507fafe07f0eb9028d91b1fa440d1/model.py#L93 here you see that the first BatchRNN's `batch_norm` parameter is hardcoded to False.

File: model.py [hunk 2]
```diff
@@ -57,6 +57,9 @@ def __init__(self, input_size, hidden_size, rnn_type=nn.LSTM, bidirectional=Fals
                             bidirectional=bidirectional, bias=False)
         self.num_directions = 2 if bidirectional else 1
 
+    def flatten_parameters(self):
+        self.rnn.flatten_parameters()
```
Comment (SeanNaren) [inline model.py]: I haven't ran this for a long time on multiple GPUs, but doesn't [this](https://github.com/SeanNaren/deepspeech.pytorch/blob/master/model.py#L66) solve the problem?

Comment (ryanleary) [reply model.py]: Doesn't seem to, no. Even with the above change, I still get the warning, but less often.

**Comment by SeanNaren** [comment]
Thanks man!
